geopyspark/geotrellis/layer.py

```diff
@@ -724,6 +725,39 @@ def __str__(self):
     def __repr__(self):
         return "RasterLayer(layer_type={})".format(self.layer_type)
 
+    def filter_by_times(self, time_intervals):
+        """Filters a ``SPACETIME`` layer by keeping only the values whose keys fall within
+        a the given time interval(s).
+
+        Args:
+            time_intervals (``[datetime.datetime]``): A list of the time intervals to query.
+                This list can have one or multiple elements. If just a single element, then only
+                exact matches with that given time will be kept. If there are multiple times given,
+                then they are each paired together so that they form ranges of time. In the case
+                where there are an odd number of elements, then the remaining time will be treated
+                as a single query and not a range.
+
+        Note:
+            If nothing intersects the given ``time_intervals``, then the returned ``RasterLayer``
+            will be empty.
```

Comment:
Just to clarify if I'm giving `time_intervals` as `[a, b, c, d]` then the query is going to be for intervals:
`[ (a,b), (b,c), (c,d)]` ?

Reply:
The intervals would be `[ (a, b), (c, d) ]`. I used the `grouped(2)` method on the Scala `Array` to group them together.